### DISCUSSION TITLE
open another usercontrol

### DISCUSSION THREAD

**Discussion by FructusS** [AvaloniaUI/Avalonia#9895] [General]
i have main window. In mainviewmodel i binding registration usercontrol. I want the registration usercontrol to change on loginusercontrol when the button is clicked. How i can do it?
![image](https://user-images.githubusercontent.com/60427370/210869916-1f77096a-13b0-4d10-ad5b-54e9c00e12aa.png)
![image](https://user-images.githubusercontent.com/60427370/210868377-35cb0d06-258b-410e-a40c-0ba9969bfaf3.png)
registration usercontrol 
![image](https://user-images.githubusercontent.com/60427370/210870390-c218ec91-e02e-48d7-9239-4be707cbf7b5.png)

**Comment by timunie** [comment]
See https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src%2FAvalonia.Samples%2FRouting%2FBasicViewLocatorSample

**Reply by timunie** [reply]
... and please let me know if anything is missing or unclear 🙂

**Reply by FructusS** [reply]
it's a little different. here the button is in the window. I want a first user control to change second user controlwhen a button is pressed in that first user control.

**Reply by timunie** [reply]
the approach is the same. You just need a way to communicate from your FirstViewModel to MainViewModel. You can for example pass it as a paramter in the constructor: 

```cs
public FirstViewModel (ViewModelBase mainViewModel) { ... }
```

or you can use a messaging API like https://learn.microsoft.com/en-us/windows/communitytoolkit/mvvm/messenger

**Reply by FructusS** [reply]
it worked. Thank you
